static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js

```diff
@@ -3,79 +3,386 @@ define([
-            var progress = $images.length * (scrollTop/scrollHeight);
-            fastdom.write(function () {
-                $scrollEl.scrollTop(Math.floor(progress - 0.01) * scrollHeight / $images.length);
-            });
+
+        // ELEMENT BINDINGS
+        this.$galleryEl = $('.js-hosted-gallery-container');
+        this.$imagesContainer = $('.js-hosted-gallery-images');
+        this.$captionContainer = $('.hosted-gallery__captions');
+        this.$captions = $('.hosted-gallery__caption', this.$captionContainer);
+        this.$scrollEl = $('.hosted-gallery__scroll-container', this.$galleryEl);
+        this.$images = $('.hosted-gallery__image', this.$galleryEl);
+        this.$progress = $('.hosted-gallery__progress', this.$galleryEl);
+        this.$border = $('.hosted-gallery__progress--border-2', this.$progress);
+        this.prevBtn = qwery('.inline-arrow-up', this.$progress)[0];
+        this.nextBtn = qwery('.inline-arrow-down', this.$progress)[0];
+        this.infoBtn = qwery('.js-gallery-caption-button', this.$captionContainer)[0];
+        this.$counter = $('.hosted-gallery__image-count', this.$progress);
+
+        // FSM CONFIG
+        this.fsm = new FiniteStateMachine({
+            initial: 'image',
+            onChangeState: function (oldState, newState) {
+                this.$galleryEl
+                    .removeClass('hosted-gallery--' + oldState)
+                    .addClass('hosted-gallery--' + newState);
+            },
+            context: this,
+            states: this.states
         });
 
-        bean.on($downArrow[0], 'click', function () {
-            var scrollTop = $scrollEl[0].scrollTop;
-            var scrollHeight = $scrollEl[0].scrollHeight;
-            var progress = $images.length * (scrollTop/scrollHeight);
-            fastdom.write(function () {
-                $scrollEl.scrollTop(Math.ceil(progress + 0.01) * scrollHeight / $images.length);
+        bean.on(this.infoBtn, 'click', this.trigger.bind(this, 'toggle-info'));
+        this.loadSurroundingImages(1, this.$images.length);
+
+        if (this.useSwipe) {
+            this.$galleryEl.addClass('use-swipe');
+            this.initSwipe();
+        } else {
+            this.$galleryEl.addClass('use-scroll');
+            this.initScroll();
+        }
+    }
+
+    HostedGallery.prototype.initScroll = function () {
+        bean.on(document.body, 'keydown', this.handleKeyEvents.bind(this));
+
+        bean.on(this.nextBtn, 'click', this.scrollTo.bind(this, 1));
+        bean.on(this.prevBtn, 'click', this.scrollTo.bind(this, -1));
+
+        bean.on(this.$scrollEl[0], 'scroll', throttle(this.fadeContent.bind(this), 20));
+    };
+
+    HostedGallery.prototype.initSwipe = function () {
+
+        var threshold, ox, dx, touchMove,
+            updateTime = 20; // time in ms
+        this.swipeContainerWidth = this.$galleryEl.dim().width;
+        this.$imagesContainer.css('width', this.$images.length + '00%');
```

Comment:
It will still work fine, this is to set the width of the containing element. Then the images all take up equal space inside, so they're each 100% width (of the screen).